### ISSUE TITLE
Residual villagers

### ISSUE THREAD

**Issue by RedDiamondBlood** [ImCoolYeah105/Mechanization#52]
I've been testing this amazing datapack and playing with the gadgets when turned my gamemode into spectator and then i saw that a lot of villagers were standing in the air. They're not a problem when they're not in front of a crafting table or something because we cannot interact with these. 
![image](https://user-images.githubusercontent.com/42066825/47732847-4ad4c380-dc67-11e8-8211-f49c50c0b95c.png)
I think they appears when we hold something from the datapack like the multimeter.
Thank you for your time.

**Comment by Forbidden-A** [comment]
Sent from Yahoo Mail for iPhone


On Tuesday, October 30, 2018, 19:21, RedDiamondBlood <notifications@github.com> wrote:


I've been testing this amazing datapack and playing with the gadgets when turned my gamemode into spectator and then i saw that a lot of villagers were standing in the air. They're not a problem when they're not in front of a crafting table or something because we cannot interact with these.

I think they appears when we hold something from the datapack like the multimeter.
Thank you for your time.

**Comment by RedDiamondBlood** [comment]
Oh I forgot to tell that i'm playing with other datapacks and i think it's the problem because these villagers don't appears when there is only mechanization on the world.

**Comment by ICY105** [comment]
What other datapacks? It's possible they're messing with the villagers in some way that Datapack Utils doesn't expect.
However, this is a tad strange because villagers are bound to a specific player, and if that player deselects an item with click detection, all bound villagers are killed.

**Comment by RedDiamondBlood** [comment]
The datapacks on my world are:
-Mechanization
-Datapack utilities
-Magic sorting system
-Math lib
-Nitwit trade
-octuple stairs
-rotten to leather
-tree capitator
-all mob heads
-anvil mending
-carpet on stairs
-chestplate + elytra
-d2g elevator
-double shulker shell
-furniture datapack
-item frame color
-soulshard
Yeah that's a lot and i should put less datapacks :/
However i think the problem could come from nitwit trades because the villagers the datapack uses are nitwits, but i'm not sure.

**Comment by Darkspyro911** [comment]
yes the mech use the village left click dection (left click dection is a bug)

On 11/1/18, RedDiamondBlood <notifications@github.com> wrote:
> The datapacks on my world are:
> -Mechanization
> -Datapack utilities
> -Magic sorting system
> -Math lib
> -Nitwit trade
> -octuple stairs
> -rotten to leather
> -tree capitator
> -all mob heads
> -anvil mending
> -carpet on stairs
> -chestplate + elytra
> -d2g elevator
> -double shulker shell
> -furniture datapack
> -item frame color
> -soulshard
> Yeah that's a lot and i should put less datapacks :/
> However i think the problem could come from nitwit trades because the
> villagers the datapack uses are nitwits, but i'm not sure.
>
> --
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/ImCoolYeah105/Mechanization/issues/52#issuecomment-435226493


-- 
Marc-Andre Maurice
mauricemarcandre@gmail.com
